values/time_test.go

```diff
@@ -80,6 +81,115 @@ func TestTime_Truncate(t *testing.T) {
 	}
 }
 
+func TestTime_Add(t *testing.T) {
+	// Note: 2020 is a leap year. Some of these tests
+	// pass through that year to test leap years operate
+	// correctly.
+	for _, tt := range []struct {
+		t    string
+		d    string
+		want string
+	}{
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "1ns",
+			want: "2019-01-01T00:00:00.000000001Z",
+		},
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "1d",
+			want: "2019-01-02T00:00:00Z",
+		},
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "1mo",
+			want: "2019-02-01T00:00:00Z",
+		},
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "2mo",
+			want: "2019-03-01T00:00:00Z",
+		},
+		{
+			t:    "2019-01-31T00:00:00Z",
+			d:    "2mo",
+			want: "2019-03-31T00:00:00Z",
+		},
+		{
+			t:    "2019-02-28T00:00:00Z",
+			d:    "2mo",
+			want: "2019-04-28T00:00:00Z",
+		},
+		{
+			t:    "2019-01-31T00:00:00Z",
+			d:    "1mo",
+			want: "2019-02-28T00:00:00Z",
+		},
+		{
+			t:    "2020-01-31T00:00:00Z",
+			d:    "1mo",
+			want: "2020-02-29T00:00:00Z",
+		},
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "1y",
+			want: "2020-01-01T00:00:00Z",
+		},
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "2y",
+			want: "2021-01-01T00:00:00Z",
+		},
+		{
+			t:    "2018-01-01T00:00:00Z",
+			d:    "1y6mo",
+			want: "2019-07-01T00:00:00Z",
+		},
+		{
+			t:    "2019-01-01T00:00:00Z",
+			d:    "1y6mo",
+			want: "2020-07-01T00:00:00Z",
+		},
+		{
+			// Not a leap year. Multiple of 100.
+			t:    "2100-01-01T00:00:00Z",
+			d:    "1y",
+			want: "2101-01-01T00:00:00Z",
+		},
+		{
+			// Not a leap year. Multiple of 100.
+			t:    "2100-01-31T00:00:00Z",
+			d:    "1mo",
+			want: "2100-02-28T00:00:00Z",
+		},
+		{
+			// Is a leap year. Multiple of 400.
+			t:    "2000-01-01T00:00:00Z",
+			d:    "1y",
+			want: "2001-01-01T00:00:00Z",
+		},
+		{
+			// Is a leap year. Multiple of 400.
+			t:    "2000-01-31T00:00:00Z",
+			d:    "1mo",
+			want: "2000-02-29T00:00:00Z",
+		},
```

Comment:
It seems like you need test cases for negative durations here?

Reply:
Added the test cases and made it work with negative values.